python/variable.cpp

```diff
@@ -182,16 +182,17 @@ void bind_init_0D_numpy_types(py::class_<Variable> &c) {
   c.def(py::init([](py::buffer &b, const std::optional<py::buffer> &v,
                     const units::Unit &unit, py::object &dtype) {
           py::buffer_info info = b.request();
-          if (info.ndim == 0) {
-            auto arr = py::array(b);
-            auto varr = v ? std::optional{py::array(*v)} : std::nullopt;
-            return doMakeVariable({}, arr, varr, unit, dtype);
-          } else if (info.ndim == 1 &&
-                     scipp_dtype(dtype) == core::dtype<Eigen::Vector3d>) {
-            return do_init_0D<Eigen::Vector3d>(
-                b.cast<Eigen::Vector3d>(),
-                v ? std::optional(v->cast<Eigen::Vector3d>()) : std::nullopt,
-                unit);
+          if (info.ndim == 1) {
+            if (scipp_dtype(dtype) == core::dtype<Eigen::Vector3d>) {
+              return do_init_0D<Eigen::Vector3d>(
+                  b.cast<Eigen::Vector3d>(),
+                  v ? std::optional(v->cast<Eigen::Vector3d>()) : std::nullopt,
+                  unit);
+            } else {
+              auto arr = py::array(b);
+              auto varr = v ? std::optional{py::array(*v)} : std::nullopt;
+              return doMakeVariable({}, arr, varr, unit, dtype);
+            }
```

Comment:
Can you explain this change? Why was this necessary? Why is `ndim == 0` not handled any more?

Reply:
Ah yes, apparently when you do `np.int64(6)` it gives ndim = 1 here. I wanted to ask you about it and then forgot

Reply:
I find numpy's handling of 0-D arrays weird. I thought there should be some test covering this?

Reply:
Yes, and what's weird is that this doesn't seem related to the changes you or I made.
But before i modified this, this test was no longer passing: https://github.com/scipp/scipp/blob/master/python/tests/test_dataset.py#L408
i don't really understand why

Reply:
Which version of numpy were you using? A change in their API around this was the reason for requiring a quite recent one.